snapd-glib/snapd-notices-monitor.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "snapd-notices-monitor.h"
+
+struct _SnapdNoticesMonitor {
+    GObject parent_instance;
+
+    SnapdClient *client;
+    GCancellable *cancellable;
+    GDateTime *last_date_time;
+    gdouble last_date_time_seconds;
+};
+
+G_DEFINE_TYPE(SnapdNoticesMonitor, snapd_notices_monitor, G_TYPE_OBJECT)
+
+static void begin_monitor(SnapdNoticesMonitor *self);
+
+static void monitor_cb(SnapdClient* source,
+                       GAsyncResult* res,
+                       SnapdNoticesMonitor *self) {
+
+    g_autoptr(GPtrArray) notices = NULL;
+    g_autoptr(GError) error = NULL;
+
+    notices = snapd_client_get_notices_finish(source, res, &error);
+
+    if (error != NULL) {
```

Comment:
Once using an internal cancellable - If error is G_IO_ERROR_CANCELLED then should just immediately return as the monitor may no longer exist.